File: hinovel-web/src/api/novel-memo.ts
import http from './http'

/**
 * 小说创作手记接口
 */
export interface NovelMemo {
    id: number
    novelId: number
    userId: number
    content: string
    createTime: string
    updateTime: string
}

/**
 * 获取小说的创作手记
 *
 * @param novelId 小说 ID
 */
export const getNovelMemo = (novelId: number) =>
    http.get<never, { data: NovelMemo | null }>(`/api/novel/memo/${novelId}`)

/**
 * 保存创作手记
 *
 * @param novelId 小说 ID
 * @param content 手记内容（Markdown 格式）
 */
export const saveNovelMemo = (novelId: number, content: string) =>
    http.put<never, { data: void }>(`/api/novel/memo/${novelId}`, {content})

File: hinovel-server/src/main/java/cn/haowl/hinovel/novel/domain/entity/NovelMemo.java
package cn.haowl.hinovel.novel.domain.entity;

import cn.haowl.hinovel.common.entity.BaseEntity;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.*;

/**
 * 小说创作手记实体。
 *
 * <p>用于记录小说创作过程中的笔记内容，支持 Markdown 格式，
 * 仅供作者个人阅读，不参与 AI 流程。</p>
 *
 * @author haowl
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@TableName("novel_memo")
public class NovelMemo extends BaseEntity {

    /**
     * 小说 ID。
     */
    private Long novelId;

    /**
     * 用户 ID。
     */
    private Long userId;

    /**
     * 手记内容（Markdown 格式）。
     */
    private String content;
}
